packages/xo-server-perf-alert/src/index.js

```diff
@@ -524,11 +524,39 @@ ${monitorBodies.join('\n')}`
                   })
                 }
 
-                result.listItem = `  * ${result.objectLink}: ${
-                  result.value === undefined
-                    ? "**Can't read performance counters**"
-                    : result.value.toFixed(1) + typeFunction.unit
-                }\n`
+                const getListItem = () => {
+                  if (result.value === undefined) {
+                    return "**Can't read performance counters**"
+                  }
+
+                  if (lcObjectType === 'vm' && definition.variableName === 'memoryUsage') {
+                    const checkManagementAgent = (vm, guestMetrics) => {
+                      if ((vm.power_state !== 'Running' && vm.power_state !== 'Paused') || guestMetrics === undefined) {
+                        return
+                      }
+
+                      const { major, minor } = guestMetrics.PV_drivers_version
+                      const hasPvVersion = major !== undefined && minor !== undefined
```

Comment:
If you're trying to replicate the value of `managementAgentDetected` from `xapi-object-to-xo`, you also need to test if `vm.metrics` is `undefined` and you need to use the `PV_drivers_detected` field as the primary detection method.